File: My neural network project #2/Web App/neuron.js

// this is the class that represent the neurons

function Neuron() {

    this.weights = [];
    this.value = 0;
    this.sum = 0;
    this.changed = false;
    this.ran = false;

    this.thresholdFunc = function() {
        this.ran = true;
        var previousValue = this.value;

        if (this.sum >= 0){
            this.value = 1;
        }
        else{
            this.value = 0;
        }

        if (previousValue != this.value){
            this.changed = true;
        }

    }
}

File: My neural network project #2/neuron.py
import random
import math
# this is the class that represent the neurons

class Neuron(object):
        ##### Code that runs #####
        def __init__(self):
                self.weights = []
                self.value = 0
                self.sum = 0
                self.changed = False
                self.ran = False

        def thresholdFunc(self):
            self.ran = True
            previousValue = self.value

            if self.sum >= 0:
                self.value = 1
            else:
                self.value = 0

            if not(previousValue == self.value):
                self.changed = True